evm-verifier/contracts/EVMProofHelper.sol

```diff
@@ -88,27 +88,21 @@ library EVMProofHelper {
         function(address,uint256,bytes memory, bytes32) internal view returns(bytes memory) getter,
         bytes32 storageRoot, uint256 slot, bytes[] memory proof, uint256 proofIdx) private view returns(bytes memory value, uint256 newProofIdx
     ) {
-        uint256 firstValue = uint256(getFixedValue(target, getter,storageRoot, slot, proof[proofIdx++]));
+        uint256 firstValue = uint256(getFixedValue(target, getter, storageRoot, slot, proof[proofIdx++]));
         if(firstValue & 0x01 == 0x01) {
             // Long value: first slot is `length * 2 + 1`, following slots are data.
-            uint256 length = (firstValue - 1) / 2;
-            value = "";
             slot = uint256(keccak256(abi.encodePacked(slot)));
```

Comment:
```suggestion
            slot = uint256(keccak256(bytes32(slot)));
```

Reply:
These are equivalent.  The `bytes32` isn't necessary either.  This was a leftover when I was making the edits and realized I was changing too much.  I kept this change due to the prior comment.

Reply:
I also wouldn't be surprised. Sadly however Solidity's `keccak256` method takes a bytes value and won't implicitly convert:

"_TypeError: Invalid type for argument in function call. Invalid implicit conversion from bytes32 to bytes memory requested. This function requires a single bytes argument. Use abi.encodePacked(...) to obtain the pre-0.5.0 behaviour or abi.encode(...) to use ABI encoding._"